### DISCUSSION TITLE
I would like to know how to add items to signedheaders.

### DISCUSSION THREAD

**Discussion by riruo** [FantasticFiasco/aws-signature-version-4#712] [Q&A] [answered]
I found this library for aws s4 authentication.

I was following along with the sample, but I need to add an optional entry other than host to the signedheaders.

However, I could not find a way to add the item in postasync.
(There was a test class, but that was incomplete as a code example)

Could you please tell me how to add arbitrary items to signedheaders to solve the above problem?

**Comment by FantasticFiasco** [comment, marked as answer]
Hi @riruo.

The API of this library is pretty much the same as the API of `HttpClient`, and if you wan't to add e.g. headers to a request you create an instance of `HttpRequestMessage` where both method and headers are specified. You then pass the request representation to the `SendAsync` overload. Will this work for you?

**Reply by riruo** [reply]
Thanks for the answer!

By re-reading the AWS SDK documentation, 
I was able to add a header to the S4 Signature with the same specifications as the HttpClient header, as you mentioned.

Thank you!
